src/combinatorics/additive/energy.lean

```diff
@@ -0,0 +1,66 @@
+/-
+Copyright (c) 2022 Yaël Dillies Ella Yu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies Ella Yu
+-/
+import data.finset.prod
+
+/-!
+# Additive energy
+
+This file defines the additive energy of two finsets of a group. This is a central quantity in
+additive combinatorics.
+-/
+
+variables {α : Type*} [decidable_eq α]
+
+namespace finset
+section has_mul
+variables [has_mul α] {s s₁ s₂ t t₁ t₂ : finset α}
+
+/-- The multiplicative energy of two finsets `s` and `t` in a group is the number of quadruples
+`(a₁, b₁, a₂, b₂) ∈ s × t × s × t` such that `a₁ * b₁ = a₂ * b₂`. -/
```

Comment:
The docstring doesn't match the definition directly - the definition filters over `(s ×ˢ s) ×ˢ t ×ˢ t` while your docstring filters over `s × t × s × t` (and you don't seem to have a lemma saying they're the same)

Reply:
I think this is a fine lie to make (and the lemma you're suggesting is useless). They are clearly isomorphic definitions and I don't want to explain precedence of `×`.

Reply:
In that case, why not just rearrange the docstring to be in the same order as the definition? This isn't just a matter of associativity or precedence

Reply:
Do you want me to write `((a₁, a₂), b₁, b₂) ∈ (s × s) × t × t`?

Reply:
I'd be happy with that, or `(a₁, a₂, b₁, b₂) ∈ s × s × t × t`

Reply:
I went for the latter.